webapp/portlet/src/main/webapp/mfa-access/components/FidoAccess.vue

```diff
@@ -0,0 +1,294 @@
+              }
+              if ('attestation' in options) {
+                makeCredentialOptions.attestation = options.attestation;
+              }
+              if ('excludeCredentials' in options) {
+                makeCredentialOptions.excludeCredentials = this.credentialListConversion(options.excludeCredentials);
+              }
+
+              console.log('sending attestation request:');
+              console.log(makeCredentialOptions);
+              if (makeCredentialOptions.excludeCredentials.length>0) {
+                //at least one authenticator is registred
+                //so we should make authentication instead of register
+                this.startAuthentication();
+              } else {
+
+                this.changeScreen('register');
+                navigator.credentials.create({
+                  'publicKey': makeCredentialOptions
+                }).then(attestation => {
+                  this.finishRegistration(attestation);
+                }).catch((err) => {
+                  this.changeScreen('error');
+                  console.error('Unable to create credentials', err);
+                });
+              }
+            });
+          } else {
+            this.changeScreen('success');
+            console.error('Error when starting registration');
+          }
+        });
+      } else {
+        this.changeScreen('error');
+        console.error('WebAuthn is not available on this browser');
+      }
+    },
+    finishRegistration(attestation) {
+      console.log('received attestation response');
+      console.log(attestation);
+      const publicKeyCredential = {};
+
+      if ('id' in attestation) {
+        publicKeyCredential.id = attestation.id;
+      }
+      if ('type' in attestation) {
+        publicKeyCredential.type = attestation.type;
+      }
+      if (!attestation.response) {
+        console.error('Make Credential response lacking response attribute');
+        return;
+      }
+
+      const response = {};
+      response.clientDataJSON = this.binToStr(attestation.response.clientDataJSON);
+      response.attestationObject = this.binToB64Str(attestation.response.attestationObject);
+
+      publicKeyCredential.response = response;
+
+      console.log(publicKeyCredential);
+      console.log(JSON.stringify(publicKeyCredential));
+      let formData = new FormData();
+      formData.append('data', JSON.stringify(publicKeyCredential));
+      formData=new URLSearchParams(formData);
+      fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/fido/finishRegistration`, {
+        method: 'POST',
+        credentials: 'include',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+          'pragma': 'no-cache',
+          'cache-control': 'no-cache',
+        },
+        body: formData
+      }).then(resp => {
+        if (resp && resp.ok) {
+          console.log('Registration success');
```

Comment:
Here we have a success case, should we display success screen ?

Reply:
Done !